application/src/main/java/org/togetherjava/tjbot/commands/moderation/attachment/AttachmentListener.java

```diff
@@ -0,0 +1,126 @@
+package org.togetherjava.tjbot.commands.moderation.attachment;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.MessageBuilder;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.dv8tion.jda.api.requests.RestAction;
+import net.dv8tion.jda.api.requests.restaction.AuditableRestAction;
+import org.jetbrains.annotations.NotNull;
+import org.jooq.tools.StringUtils;
+import org.togetherjava.tjbot.commands.MessageReceiverAdapter;
+import org.togetherjava.tjbot.config.Config;
+import org.togetherjava.tjbot.moderation.ModAuditLogWriter;
+
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+/**
+ * Listener that receives all sent messages in every channel and checks if the message has any
+ * blacklisted file attached.
+ * <p>
+ * If there was a blacklisted file attached, delete the message and send the user a dm, telling what
+ * they did wrong and also inform the mods.
+ */
+public class AttachmentListener extends MessageReceiverAdapter {
+
+    private final Config config;
+
+    private final ModAuditLogWriter modAuditLogWriter;
+
+    /**
+     * Creates a AttachmentListener to receive all message sent in any channel.
+     *
+     * @param config to find the blacklisted media attachments
+     * @param modAuditLogWriter to inform the mods about the suspicious attachment
+     */
+    public AttachmentListener(@NotNull Config config,
+            @NotNull ModAuditLogWriter modAuditLogWriter) {
+        super(Pattern.compile(".*"));
+        this.config = config;
+        this.modAuditLogWriter = modAuditLogWriter;
+    }
+
+    @Override
+    public void onMessageReceived(@NotNull MessageReceivedEvent event) {
+        if (event.getAuthor().isBot() || event.isWebhookMessage()) {
+            return;
+        }
+        if (attachmentContainsBlacklistedFileExtension(event)) {
+            deleteMessage(event).flatMap(any -> dmUser(event)).queue();
+            warnMods(event);
+        }
+    }
+
+    private AuditableRestAction<Void> deleteMessage(@NotNull MessageReceivedEvent event) {
```

Comment:
u don't need the whole event here, make it a `Message`

Reply:
in my last PR @Zabuzard requested this behavior --> https://github.com/Together-Java/TJ-Bot/pull/518#discussion_r949982321

Reply:
i deleted the method  but i added a nice wrapper deleteMessageAndDmUser which calls  event.getMessage().delete().flatMap(any -> dmUser(event)).queue(); in my opinion it is much cleaner and easier to read

Reply:
a method should not do more than one thing

Reply:
in my last PR @Zabuzard requested this behavior --> https://github.com/Together-Java/TJ-Bot/pull/518#discussion_r949982321

should i just  extract two methods with separated queues?

Reply:
Regardless of how u code it, you have to flatmap these actions in one chain. calling `queue` twice is stupid and blocks the thread unecessarily.

Also, you dont want to DM the user if the deletion failed for some reason - the chain is important. But I agree that the method name is kinda awkward, it should instead be a more abstract name that represents the general topic of whats going on, such as `handleBadMessage`.

Reply:
i totally agree with the ``queue`` but with the naming ``handleBadMessage`` you always have to look into the detail implementation. For me the best solution would be

```Java
private void handleBadMessage(@NotNull MessageReceivedEvent event) {
       deleteMessage();
       dmUser();
       warnMods()

    }
```
But it is just my opinion 😄

Reply:
thats fine. but it has to be a single action queue, like `first.then(second).then(third)` (but with JDAs RestAction)